UnityProject/Storybook/Assets/Code/Scripts/Game/Player/PlayerMover.cs

```diff
@@ -0,0 +1,338 @@
+    [PunRPC]
+    public void SwitchCameraRoom(Vector3 newCamPos, Quaternion newCamRot)
+    {
+        Camera.main.transform.position = newCamPos;
+        Camera.main.transform.rotation = newCamRot;
+    }
+
+    /// <summary>
+    /// Sends the current room data to the clients
+    /// </summary>
+    /// <param name="isNorthDoorActive">True if the north door is active for the current room, false otherwise</param>
+    /// <param name="isEastDoorActive">True if the east door is active for the current room, false otherwise</param>
+    /// <param name="isSouthDoorActive">True if the south door is active for the current room, false otherwise</param>
+    /// <param name="isWestDoorActive">True if the west door is active for the current room, false otherwise</param>
+    /// <param name="roomType">The type of the room (MapManager.RoomType)</param>
+    [PunRPC]
+    public void SendRoomData(bool isNorthDoorActive, bool isEastDoorActive, bool isSouthDoorActive, bool isWestDoorActive, int roomType)
+    {
+        m_currentRoomData = new RoomData(isNorthDoorActive, isEastDoorActive, isSouthDoorActive, isWestDoorActive, (MapManager.RoomType) roomType);
+    }
+
+    /// <summary>
+    /// Sends the location of the current room to the clients
+    /// </summary>
+    /// <param name="roomLocX">The x-coordinate of the current room</param>
+    /// <param name="roomLocY">The y-coordinate of the current room</param>
+    [PunRPC]
+    public void SendRoomLoc(int roomLocX, int roomLocY)
+    {
+        m_currentRoomLoc = new Location(roomLocX, roomLocY);
+    }
+
+    /// <summary>
+    /// Called on clients to create a room on the server
+    /// </summary>
+    /// <param name="roomLocX">The x-coordinate of the room to create</param>
+    /// <param name="roomLocY">The y-coordinate of the room to create</param>
+    /// <param name="doorIndex">The door that was selected by the client</param>
+    [PunRPC]
+    public void CreateRoomOnMaster(int roomLocX, int roomLocY, int doorIndex)
+    {
+        Door selectedDoor = m_currentRoom.RoomDoors[doorIndex];
+
+        // If the selected door's room has not been spawned, create the room
+        if (!selectedDoor.IsDoorRoomSpawned)
+        {
+            CreateNewRoom(roomLocX, roomLocY);
+            selectedDoor.SetIsDoorRoomSpawned(true);
+        }
+
+        StartCoroutine(MoveThroughDoor(selectedDoor));
+    }
+
+    /// <summary>
+    /// The photon player id of the current leader
+    /// </summary>
+    [SyncProperty]
+    public int LeaderId
+    {
+        get { return m_leaderId; }
+        set
+        {
+            m_leaderId = value;
+            PropertyChanged();
+        }
+    }
+
+    /// <summary>
+    /// True when the players are waiting in the center of the room, false while they are moving
+    /// If it is true, the buttons will appear to the leader, if it false, they will not appear
+    /// </summary>
+    [SyncProperty]
+    public bool AreButtonsEnabled
```

Comment:
IF this is test code then it should be in a child class, not in the class itself. Only put test code on classes we can delete.